impl/src/main/java/org/ehcache/internal/store/offheap/AbstractOffHeapStore.java

```diff
@@ -868,7 +868,16 @@ public boolean hasNext() throws CacheAccessException {
       final Map.Entry<K, OffHeapValueHolder<V>> thisEntry = next;
       advance();
 
-      setAccessTimeAndExpiry(thisEntry.getKey(), thisEntry.getValue(), timeSource.getTimeMillis());
+      final K key = thisEntry.getKey();
+      backingMap().computeIfPresent(key, new BiFunction<K, OffHeapValueHolder<V>, OffHeapValueHolder<V>>() {
```

Comment:
This change will update the value holder in store but we still return the old one with one less hit and older last access time. Since we are computing anyway, we could update the one we hold with the latest returned, and only return the old one if the actual store no longer holds this mapping.